isis/tests/TestUtilities.cpp

```diff
@@ -65,31 +65,36 @@ namespace Isis {
       PvlGroup group1,
       PvlGroup group2) {
     if (group1.name() != group2.name()) {
+      ADD_FAILURE();
       return ::testing::AssertionFailure() << "PvlGroup " << group1_expr
           << " has name (" << group1.name().toStdString() << ") and PvlGroup "
```

Comment:
So does ::testing::AssertionFailure just not work for asserting failures we need to add the `ADD_FAILURE` line?

Reply:
yeah ::testing::AssertionFailure won't work for asserting a failure with out adding the ``ADD_FAILURE`` line since by default ::testing::AssertionFailure will return True if something has failed.